app/models/user.rb

```diff
@@ -7,14 +9,43 @@ class User < ApplicationRecord
     presence: true,
     uniqueness: {case_sensitive: false},
     length: {maximum: 255}
-  validates :password, presence: true, length: {minimum: 6}
+  validates :password, presence: true, length: {minimum: 6}, on: :create
   validates :name, presence: true, length: {maximum: 50}
   validates :sex, presence: true
 
   has_secure_password
 
   before_save :email_downcase
 
+  class << self
+    def digest string
+      if ActiveModel::SecurePassword.min_cost
+        cost = BCrypt::Engine.MIN_COST
+      else
+        BCrypt::Engine.cost
+      end
+      BCrypt::Password.create string, cost: cost
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    remember_token = User.new_token
```

Comment:
cái này e xóa nhầm :v bên trên có method getter remember_token nên phải dùng instance_variable. Dùng instance_variable vì remember_token chỉ thuộc đối tượng User và nó được sử dụng trong toàn bộ class User này